src/Simplic.Studio.Tenant.UI/OrganizationViewModel.cs

```diff
@@ -0,0 +1,264 @@
+            }
+
+            this.IsCreate = true;
+        }
+
+        /// <summary>
+        /// Constructor for the view model
+        /// </summary>
+        /// <param name="id">Tenant ID</param>
+        /// <param name="name">Tenant name</param>
+        /// <param name="matchCode">Tenant match code</param>
+        /// <param name="subOrganizationCount">Tenant suborganizations count</param>
+        /// <param name="subOrganizations">Tenant suborganizations IDs collection</param>
+        /// <param name="cloudOrganizationId">Tenant cloud organization id</param>
+        /// <param name="isActive">Tenant active flag</param>
+        /// <param name="isGroup">Tenant group flag</param>
+        public OrganizationViewModel(Guid id, string name, string matchCode, int subOrganizationCount,
+            IList<Guid> subOrganizations, Guid? cloudOrganizationId, bool isActive, bool isGroup) : this()
+        {
+            if (this.Model == null)
+            {
+                this.model = new Organization
+                {
+                    Id = id,
+                    Name = name,
+                    MatchCode = matchCode,
+                    IsActive = isActive,
+                    SubOrganizations = subOrganizations,
+                    CloudOrganizationId = cloudOrganizationId
+                };
+
+                IsCreate = true;
+            }
+
+        }
+
+        /// <summary>
+        /// Copy constructor
+        /// </summary>
+        /// <param name="org"></param>
+        public OrganizationViewModel(Organization org) : this(org.Id, org.Name, org.MatchCode, org.SubOrganizationCount,
+            org.SubOrganizations, org.CloudOrganizationId, org.IsActive, org.IsGroup)
+        {
+            this.model = org;
+            this.IsCreate = false;
+        }
+
+        /// <summary>
+        /// Gets all user assigned organizations.
+        /// </summary>
+        /// <param name="user">Selected user.</param>
+        /// <returns>List of all user assigned organizations.</returns>
+        public IEnumerable<Organization> GetUserOrganizations(User.User user)
+        {
+            return organizationService.GetByUserId(user.Ident);
+        }
+
+        /// <summary>
+        /// Saves <see cref="Organization"/> instance.
+        /// </summary>
+        /// <returns>Wether the safe was succesfull.</returns>
+        public bool SaveOrganization()
+        {
+            if (Model != null)
+            {
+                if (IsCreate)
+                {
+                    var newTenant = new Organization
+                    {
+                        Id = Model.Id,
+                        Name = Model.Name,
+                        MatchCode = Model.MatchCode,
+                        IsActive = Model.IsActive,
+                        SubOrganizations = Model.SubOrganizations,
+                        CloudOrganizationId = Model.CloudOrganizationId
+                    };
```

Comment:
why creating a new object here? Especially when using the model for non new cases